python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+            inputDict[component] = componentDict
+
+        inputYAML.addInput(yaml.dump(inputDict))
+        inputYAML.addInput("Identity:\n  Type:  Identity\n")
+
+        return inputYAML
+
+    def _addObjects(self, wcsf, inputCatalogRefs):
+        """Add science sources to the wcsfit.WCSFit object.
+
+        Parameters
+        ----------
+        wcsf : `wcsfit.WCSFit`
+            WCS-fitting object
+        inputCatalogRefs : `list`
+            List of DeferredDatasetHandles pointing to visit-level source
+            tables
+        """
+        for inputCatalogRef in inputCatalogRefs:
+            visit = inputCatalogRef.dataId['visit']
+
+            inputCatalog = inputCatalogRef.get(parameters={'columns': self.columns})
+            detectors = set(inputCatalog['detector'])
+
+            for detector in detectors:
+                detectorSources = inputCatalog[inputCatalog['detector'] == detector]
+
+                extensionIndex = self.extensionInfo[(self.extensionInfo['visit'] == visit)
+                                                    & (self.extensionInfo['detector'] == detector)].index[0]
+                sourceCat = detectorSources[self.sourceIndices[extensionIndex]]
+
+                d = {"x": sourceCat['x'].to_numpy(),
+                     "y": sourceCat['y'].to_numpy(),
+                     "xCov": sourceCat['xErr'].to_numpy()**2,
+                     "yCov": sourceCat['yErr'].to_numpy()**2,
+                     "xyCov": np.zeros(len(sourceCat))}
+                # TODO: add correct xyErr if DM-7101 is ever done
+
+                wcsf.setObjects(extensionIndex, d, 'x', 'y', ['xCov', 'yCov', 'xyCov'])
+
+    def _addRefObjects(self, wcsf):
+        """Add reference sources to the wcsfit.WCSFit object.
+
+        Parameters
+        ----------
+        wcsf : `wcsfit.WCSFit`
+            WCS-fitting object
+        """
+        extensionIndex = self.extensionInfo[self.extensionInfo['extensionType'] == 'REFERENCE'].index[0]
+
+        if self.config.useProperMotion:
+            wcsf.setObjects(extensionIndex, self.refObjects, 'ra', 'dec', ['raCov', 'decCov', 'raDecCov'],
+                            pmDecKey='decPM', pmRaKey='raPM', parallaxKey='parallax', pmCovKey='fullCov',
+                            pmCov=self.refCovariance)
+        else:
+            wcsf.setObjects(extensionIndex, self.refObjects, 'ra', 'dec', ['raCov', 'decCov', 'raDecCov'])
+
+    def _makeAfwWcs(self, mapDict, centerRA, centerDec, doNormalizePixels=False, xScale=1, yScale=1):
+        """Make an `lsst.afw.geom.SkyWcs` from a dictionary of mappings
+
+        Parameters
+        ----------
+        mapDict : `dict`
+            Dictionary of mapping parameters
+        centerRA : `lsst.geom.Angle`
+            RA of the tangent point
+        centerDec : `lsst.geom.Angle`
+            Declination of the tangent point
+        doNormalizePixels : `bool`
+            Whether to normalize pixels so that range is [-1,1]
```

Comment:
I just noticed this, but all of your method docstrings should be sentences, with punctuation.

https://developer.lsst.io/python/numpydoc.html#parameters